lib/ansible/playbook/role/requirement.py

```diff
@@ -190,6 +190,17 @@ def scm_archive_role(src, scm='git', name=None, version='HEAD'):
         if rc != 0:
             raise AnsibleError ("- command %s failed in directory %s (rc=%s)" % (' '.join(clone_cmd), tempdir, rc))
 
+        if scm == 'git' and version:
+            checkout_cmd = [scm, 'checkout', version]
+            with open('/dev/null', 'w') as devnull:
+                try:
+                    popen = subprocess.Popen(checkout_cmd, cwd=os.path.join(tempdir, name), stdout=devnull, stderr=devnull)
+                except (IOError, OSError):
+                    raise AnsibleError("error executing: %s" % " ".join(checkout_cmd))
+                rc = popen.wait()
+            if rc != 0:
```

Comment:
The approach in the modification regarding testing `rc` was "adopt-and-go" with existing coding conventions.  Shifting those conventions for enhanced zen may be desired, but the scope of doing so would be larger than this PR.  Considering occurences in the ansible code base of `rc != 0`:
```
~/git/ansible$ find . -type f | xargs -I FILE grep "rc != 0" FILE /dev/null | wc -l
353
```

Making this change may or may not be desired, if so might it best be undertaken in its own PR?